app/assets/javascripts/components/edit_campaign_page.js.jsx

```diff
@@ -0,0 +1,94 @@
+CheckBox = React.createClass({
+  getInitialState() {
+    return (
+      { checked: this.props.checked }
+    )
+  },
+
+  handleChange() {
+    checked = React.findDOMNode(this.refs.checkbox).checked
+
+    data = {}
+    data[this.props.name] = checked
+    this.props.onChange(data);
+    this.setState( {checked: !this.state.checked } )
+  },
+
+  render() {
+    return (
+      <div className="checkbox">
+        <label htmlFor="campaign_page_{this.props.label}" className="control-label">
+          <input type="checkbox" checked={this.state.checked } onChange={ this.handleChange } ref="checkbox" /> {this.props.label}
+        </label>
+      </div>
```

Comment:
@osahyoun I think this is what @EricBoersma was talking about for 'HTML inline with the framework.' To me the issue isn't that it's not in rails views, but rather that the templates don't exist in their own files. Can we put the JSX in their own template files?